### PR TITLE
Test the solver in a number of uncomfortably exciting cases

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: physics/euler_solver_test.cpp [hunk 1]
```diff
@@ -535,6 +536,290 @@ TEST_F(EulerSolverTest, ДжанибековEffect) {
...
+        actual_attitude(e3),
+        Componentwise(VanishesBefore(1, 2),
+                      VanishesBefore(1, 1),
+                      AlmostEquals(expected_attitude(e3).coordinates().z, 0)));
+  }
+  // A general body that has an angular momentum very close to the third
+  // principal axis.
+  {
+    Solver::AngularMomentumBivector const initial_angular_momentum(
+        {std::numeric_limits<double>::min() * SIUnit<AngularMomentum>(),
```
Comment (eggrobin) [inline physics/euler_solver_test.cpp]: Does this go through the `ṁ_is_zero` path or the normal one? What happens with `denorm_min`?

File: physics/euler_solver_test.cpp [hunk 2]
```diff
@@ -535,6 +536,290 @@ TEST_F(EulerSolverTest, ДжанибековEffect) {
...
+            expected_angular_frequency * 10 * Second, initial_angular_momentum);
+
+    EXPECT_THAT(actual_angular_momentum,
+                AlmostEquals(initial_angular_momentum, 0));
+    EXPECT_THAT(actual_angular_velocity,
+                AlmostEquals(expected_angular_velocity, 0));
+    EXPECT_THAT(actual_attitude(e1), AlmostEquals(expected_attitude(e1), 29));
+    EXPECT_THAT(actual_attitude(e2), AlmostEquals(expected_attitude(e2), 68));
+    EXPECT_THAT(actual_attitude(e3), AlmostEquals(expected_attitude(e3), 6));
+  }
```
Comment (eggrobin) [inline physics/euler_solver_test.cpp]: Add a test with `m` somewhere in the 2-3 plane (in the 2 or 3 direction) for I2=I3.

File: physics/euler_solver_body.hpp [hunk 3]
```diff
@@ -166,11 +170,9 @@ EulerSolver<InertialFrame, PrincipalAxesFrame>::EulerSolver(
       } else {
         σʺB₃₁_ = B₃₁_;
       }
-      // Not quite an elliptic integral characteristic, but we'll stick to that
-      // notation.
-      n_ = G² * G² / (B₂₁² * B₂₃²);
-      ψ_Π_offset_ = (-ν_ + n_ * std::log(n_ * Sinh(-ν_) - Cosh(-ν_)) * Radian);
-      ψ_Π_multiplier_ = Δ₂ / (λ_ * I₂ * G_ * (1 - n_ * n_));
+      // Δ₂ shows up in the multiplier, so things simplify tremendously.
```
[the remark this answers is not in the document]
Comment (pleroy) [reply physics/euler_solver_body.hpp]: As discussed, added a comment.

File: physics/euler_solver_test.cpp (hunk 1 above)
Comment (pleroy) [reply physics/euler_solver_test.cpp]: It goes through the `ṁ_is_zero` path.  `ṁ` is actually not 0, but `NormalizeOrZero` is not safe in the face of underflows.  The same, of course, happens with `denorm_min`.

Comment (pleroy) [reply physics/euler_solver_test.cpp]: Added a test that exercises the check.

**Pull request merged**